### DISCUSSION TITLE
when hitting 'cast' the pop up window does not show white list page

### DISCUSSION THREAD

**Discussion by otaolafranc** [hensm/fx_cast#298] [Q&A]
Hello,
I tried the addon with youtube, it worked correctly, (I also installed bridge) but when doing with other random pages over internet when I hit cast and then the pop up window showing the chromcast equipment shows, it does show the possibility to whitlist the page and obviously it does not allow me to cast to my projector.
I am missing something? great work in this extension, I must say that I am quite off that firefox does not have this option by default...

**Comment by hensm** [comment]
Whitelist suggestions are only present for a small number of known sites (like YouTube). You can still whitelist a site from the toolbar button context menu or by manually entering a match pattern on the extension settings page.

This will let you cast on sites that have built in Chromecast support. If you want to mirror any random web page, you need to enable screen mirroring in the extension settings and you'll then get a screen option in the cast popup. The performance is not great for video streaming though.

**Comment by otaolafranc** [comment]
Hello @hensm  thanks for the answer, I have added the site to the white list, refresh it, but if I click in any place with right click I have the cast option but I can not select my projector (I have it on), and when I do click right over the video itself, it does not show any menu, and with shift+ right click it opens a menu, but it does not show cast...
any ideas?

**Reply by hensm** [reply]
Not all videos are cast-able. The cast menu item will only show up if the video has a routable src URL. The media casting works by sending a URL to the Chromecast and then it loads it independently. Some sites load videos indirectly in chunks or via JavaScript which means there's no URL to send.

The cast menu item on the page does the same thing as clicking on the toolbar button and will only show you the available, detected cast methods for the current page. If there's no built-in Chromecast support on that site and screen mirroring is disabled, you'll only be able to use it as a remote control.

**Reply by otaolafranc** [reply]
Hello, not sure what to say, outside that in chrome it shows the possibility to cast :/

**Reply by hensm** [reply]
Which site is this? If it has built-in Chromecast support (a cast button on the player or somewhere in the page), it might just not be compatible. If it doesn't have built-in Chromecast support, Chrome is probably showing tab/screen mirroring.

**Reply by otaolafranc** [reply]
it is a streaming page so I don't think I can share it here... the streaming icon appears inside the video player itself (next to full screen, settings, etc...), in chrome I was not casting the entire page but only the video itself.

**Reply by hensm** [reply]
It's probably just not compatible for some reason, then. If you want, you can make a compatibility report [here](https://github.com/hensm/fx_cast-compat) and I can look into it further.
